src/main/java/com/puppycrawl/tools/checkstyle/JavaAstVisitor.java

```diff
@@ -1742,6 +1760,216 @@ public DetailAstImpl visitPrimitivePrimary(JavaLanguageParser.PrimitivePrimaryCo
         return dot;
     }
 
+    @Override
+    public DetailAstImpl visitTemplateArgument(JavaLanguageParser.TemplateArgumentContext ctx) {
+        return Objects.requireNonNullElseGet(visit(ctx.template()),
+                () -> buildSimpleStringTemplateArgument(ctx));
+    }
+
+    /**
+     * Builds a simple string template argument AST, which basically means that we
+     * transform a string literal into a string template AST because it is a
+     * string template argument.
+     *
+     * @param ctx the TemplateArgumentContext to build AST from
+     * @return DetailAstImpl of string template argument
+     */
+    private static DetailAstImpl buildSimpleStringTemplateArgument(
+            JavaLanguageParser.TemplateArgumentContext ctx
+    ) {
+        final int startColumn = ctx.start.getCharPositionInLine();
+        final int endColumn = startColumn + ctx.getText().length() - 1;
```

Comment:
is this tab safe? We have a property that declares how long a tab is in size. I thought the AST respected that.

Do we have a test that shows tab usage?

Reply:
We do now, done